src/weights.jl

```diff
@@ -682,6 +682,31 @@ function mean(A::AbstractArray, w::UnitWeights; dims::Union{Colon,Int}=:)
     return mean(A, dims=dims)
 end
 
+"""
+    mean(f, A::AbstractArray, w::AbstractWeights[; dims])
+
+Compute the weighted mean of array `A`, after transforming it'S
+contents with the function `f`, with weight vector `w` (of type
+`AbstractWeights`). If `dims` is provided, compute the
+weighted mean along dimension `dims`.
+
+# Examples
+```julia
+n = 20
+x = rand(n)
+w = rand(n)
+mean(√, x, weights(w))
+```
+"""
+mean(f, A, w::AbstractWeights; kwargs...) =
+    mean(broadcast(f, A), w; kwargs...)
```

Comment:
Like I said, check [the method for weighted `sem`](https://github.com/JuliaStats/StatsBase.jl/blob/0e8e4aaaafdbacf17424e93b48e00c03674885f4/src/scalarstats.jl#L509) to see how to use `Broadcasting.broadcasted` correctly. `broadcast` is not the same thing as `Broadcasting.broadcasted`. The latter is lazy.